src/test/java/com/endava/internship/collections/StudentMapTest.java

```diff
@@ -0,0 +1,271 @@
+                () -> assertThat(studentMap.size()).isEqualTo(2)
+        );
+    }
+
+    @Test
+    public void throwException_whenGetNotStudentKey(){
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> studentMap.get("Wrong type"));
+    }
+
+    @Test
+    public void throwException_whenRemoveNotStudentKey(){
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> studentMap.remove("Wrong type"));
+    }
+
+    @Test
+    public void isEmpty_whenCleared(){
+        Student anotherStudent = new Student("another student", LocalDate.now(), "");
+        studentMap.put(dummyStudent, 1);
+        studentMap.put(anotherStudent, 2);
+
+        studentMap.clear();
+
+        assertThat(studentMap.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void isEmpty_whenPutAllEmptyMap(){
+        studentMap.putAll(new HashMap<>());
+
+        assertThat(studentMap.isEmpty()).isTrue();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentList")
+    public void sizeIsRight_whenPutAllMap(List<Student> studentList){
+        Map<Student, Integer> map = new HashMap<>();
+        for (Student student : studentList) {
+            map.put(student, 1);
+        }
+        studentMap.putAll(map);
+
+        assertThat(studentMap.size()).isEqualTo(map.size());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentList")
+    public void keySetSizeIsRight(List<Student> studentList){
+        for (Student student : studentList) {
+            studentMap.put(student, 1);
+        }
+
+        assertThat(studentMap.keySet().size()).isEqualTo(studentList.size());
+    }
+    @ParameterizedTest
+    @MethodSource("provideStudentList")
+    public void valuesCollectionSizeIsRight(List<Student> studentList){
+        for (Student student : studentList) {
+            studentMap.put(student, 1);
+        }
+
+        assertThat(studentMap.values().size()).isEqualTo(studentList.size());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentList")
+    public void EntrySetSizeIsRight(List<Student> studentList){
+        for (Student student : studentList) {
+            studentMap.put(student, 1);
+        }
+
+        assertThat(studentMap.entrySet().size()).isEqualTo(studentList.size());
+    }
+
+    @Test
+    public void rehashingDoesNotAffectMapContents(){
+        int expectedLength = 100;
+        for(int i = 0; i < expectedLength; ++i){
+            studentMap.put(new Student("Student " + i, LocalDate.now(), ""), i);
+        }
+
+        assertThat(studentMap.size()).isEqualTo(expectedLength);
+    }
+    public static Stream<Arguments> provideStudentList(){
+        List<Student> studentList = new ArrayList<>();
+        for(int i = 0; i < 5; ++i){
+            studentList.add(new Student("Student " + i, LocalDate.now(), ""));
+        }
+        return Stream.of(
+                Arguments.of(studentList)
+        );
+
+    }
+}
```

Comment:
Small note, each file should end with an empty line to eliminate this red symbol.